Improve migration guide (#490)
cylc/cylc-doc: src/7-to-8/major-changes/cli.rst

    source directory.
``cylc submit``
  - It is no longer possible to submit a job from outside of a workflow.
``cylc warranty``
  - The Cylc license remains unchanged from Cylc 7.

.. rubric:: Commands that have been replaced:

``cylc conditions``
  - See the `license`_ file for conditions of usage, or ``cylc help license``
  - The Cylc license remains unchanged from Cylc 7.
``cylc get-config``,
  - Replaced by ``cylc config``.
``cylc get-*-config``
  - (Where ``*`` is ``site``, ``suite`` or ``global``)
  - Replaced by ``cylc config``.
``cylc graph-diff``
  - Replaced by ``cylc graph <flow1> --diff <flow2>``
``cylc insert``
  - Task insertion is now automatic, use ``cylc trigger``.
``cylc monitor``
  - There is now a new more powerful terminal user interface (TUI).
  - Try ``cylc tui``.
``cylc print``
  - Equivalent to ``cylc scan --states=all``.
``cylc reset``
  - It is no longer possible to manually change a task's state.
  - You can, however, override the outputs the task generated which has a
    similar effect with ``cylc set-outputs``.
``cylc restart``
  - Replaced by ``cylc play``.
``cylc run``
  - Replaced by ``cylc play``.
``cylc spawn``

>   - Spawning is now performed automatically, use ``cylc trigger`` to run a task.

Reviewer: ```suggestion
  - Spawning is now performed automatically, on demand. Use ``cylc trigger`` to run
    a task, or ``cylc set-outputs`` to spawn tasks that depend on specified outputs.
```

``cylc suite-state``

>   - Renamed as ``cylc workflow-state``.

Reviewer: ```suggestion
  - Renamed to ``cylc workflow-state``.
```

Author: ["Renamed as" seems to be more favoured](https://books.google.com/ngrams/graph?content=renamed+as%2Crenamed+to&year_start=1800&year_end=2019&corpus=26&smoothing=3&direct_url=t1%3B%2Crenamed%20as%3B%2Cc0%3B.t1%3B%2Crenamed%20to%3B%2Cc0)

Reviewer: Actually, I agree, on reflection.

.. rubric:: Commands that have changed:

``cylc hold``
  - Now used on tasks only; use ``cylc pause`` to pause an entire workflow
    (i.e. to halt all job submissions).
``cylc release``

>   - Now used on tasks only; use ``cylc play`` to resume a paused workflow.

Reviewer: ```suggestion
  - Now used only to release held tasks; use ``cylc play`` to resume a paused workflow.
```

>   - We no longer include a command for locating this documentation.

Reviewer: Commands like `cylc documentation` and `cylc checkpoint` that have been removed should say "removed" as the first sub-bullet point. (The intro says "changed or removed" but then the reader has to figure out which is which).

Author: I've split them into 3 lists:
- commands that have been removed entirely
- those which have been replaced
- those which have changed

``cylc edit``
  - Use a text editor to edit the workflow configuration file.
``cylc hold``
  - Now used on tasks only; use ``cylc pause`` to pause an entire workflow

>     (halt all job submissions).

Reviewer: ```suggestion
    (i.e. to halt all job submissions).
```

   task.cycle
   cycle/task

In Cylc 8 the former is now deprecated, and the latter has been extended to
provide a unique identifier for all workflows, cycles, tasks and jobs using a

> standardised format:

Reviewer: ```suggestion
standard format:
```
(Not sure it has been "standardised" as such)
